### PR TITLE
Support LSP Inlay Hints in protocol specification

### PR DESCRIPTION
**Pull request by nordlow**
See https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#inlayHint

### CODE REVIEW

File: protocol/source/served/lsp/protocol.d [hunk 1]
```diff
@@ -3085,6 +3086,43 @@ struct ReferenceParams
...
+struct InlayHint
+{
+    Position position;
+    Variant!(string, InlayHintLabelPart[]) label;
+    @serdeOptional Optional!InlayHintKind kind;
+    @serdeOptional Optional!(TextEdit[]) textEdits;
+    @serdeOptional Variant!(void, string, MarkupContent) tooltip;
+    @serdeOptional Optional!(bool) paddingLeft;
+    @serdeOptional Optional!(bool) paddingRight;
+    // TODO: @serdeOptional Optional!(LSPAny) data;
```
Comment (nordlow) [inline protocol/source/served/lsp/protocol.d:3107]: LSPAny is referenced in the `metaModel.json` but couldn't find a corresponding reference in `serve-d`. I've verified that this PR works without it.

Comment (WebFreak001) [reply protocol/source/served/lsp/protocol.d:3107]: you can use OptionalJsonValue here instead

Comment (WebFreak001) [reply protocol/source/served/lsp/protocol.d:3107]:
```suggestion
    OptionalJsonValue data;
```

**Review by WebFreak001 — CHANGES_REQUESTED** [review]

File: protocol/source/served/lsp/protocol.d [hunk 2]
```diff
@@ -3151,6 +3189,14 @@ struct DocumentSymbolOptions
 	@serdeOptional Optional!string label;
 }
 
+@serdeFallbackStruct
+struct InlayHintProvider
+{
+	mixin WorkDoneProgressOptions;
+
+	@serdeOptional Optional!string label;
```
[the remark this answers is not in the document]
Comment (nordlow) [reply protocol/source/served/lsp/protocol.d]: Fixed.

File: protocol/source/served/lsp/protocol.d (hunk 1 above)
Comment (nordlow) [reply protocol/source/served/lsp/protocol.d]: Fixed.

**Comment by nordlow** [comment]
Ready for another round of reviews.

**Review by WebFreak001 — APPROVED** [review]

**Comment by WebFreak001** [comment]
fixed a bug in my suggestion and pushed to your branch directly

**Pull request merged**